common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -607,6 +590,25 @@ protected void wakeup(boolean inEventLoop) {
         }
     }
 
+    @Override
+    final void executeScheduledRunnable(final Runnable runnable, boolean isAddition, long deadlineNanos) {
+        // Don't wake up for removals
```

Comment:
either remove the comment or also add that it will do no wakeups with `beforeFutureTaskScheduled(...)` returns false